app/services/CheckoutService.scala

```diff
@@ -280,10 +281,17 @@ class CheckoutService(identityService: IdentityService,
 
     val contractEffective = subscription.termEndDate
     val customerAcceptance = nextFridayFrom(contractEffective)
-    def addPlan = {
+    def addPlan(contact: Contact) = {
       val newRatePlan = RatePlan(renewal.plan.id.get, None)
       val renewCommand = Renew(subscription.id.get, subscription.startDate, NonEmptyList(newRatePlan), contractEffective, customerAcceptance)
-      zuoraService.renewSubscription(renewCommand)
+      val validPromotion = for {
+        code <- renewal.promoCode
+        deliveryCountryString <- contact.mailingCountry
+        deliveryCountry <- CountryGroup.countryByName(deliveryCountryString)
+        validPromotion <- promoService.validate[NewUsers](code, deliveryCountry, renewal.plan.id).toOption // would be better to have valid/invalid/tracking
```

Comment:
Agreed, we should do the conversion to convert the promotion into a Tracking promo code if it's 'valid' but not for the chosen rate plan.